### PR TITLE
Bugfix, filter defaults and readme changes.

### PR DESCRIPTION
**Pull request by MicahZoltu**
Fixes a bug where filters weren't being passed to RPC correctly.

Changes default to no filters (meaning no logs) rather than defaulting to a filter that includes all logs.

Updates readme with full example.

### CODE REVIEW

**Comment by MicahZoltu** [comment]
Rebasing now...

File: README.md [hunk 1]
```diff
@@ -1,54 +1,101 @@
...
+//    .catch(error => callback(error, undefined));
+//}
+function getLogs(filterOptions: FilterOptions): Promise<Log[]> {
+    return fetch("http://localhost:8545", {
+        method: "POST",
+        headers: new Headers({"Content-Type": "application/json"}),
+        body: { jsonrpc: "2.0", id: 1, method: "eth_getLogs", params: [filterOptions] }
+    }).then(response => response.json());
+}
+//function getLogsCallbackStyle(filterOptions: FilterOptions): Promise<Log[]> {
```
Comment (tinybike) [inline README.md]: Should this use a callback instead of a promise (per the name)?

**Comment by tinybike** [comment]
Edited the callback-style examples in README (hopefully I didn't botch the types...).  Otherwise looks good, merging.

File: README.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ function getBlockByHash(hash: string): Promise<Block|null> {
         body: { jsonrpc: "2.0", id: 1, method: "eth_getBlockByHash", params: [hash, false] }
     }).then(response => response.json());
 }
-//function getBlockByHashCallbackStyle(hash: string, callback: (block: Block|null) => void): void {
+//function getBlockByHashCallbackStyle(hash: string, callback: (error: Error, block?: Block) => void): void {
```
Comment (MicahZoltu) [inline README.md]: Ah, good catch on this.  It should actually be `error?: Error, block?: Block` (both are optional).

Comment (MicahZoltu) [inline README.md]: I lied, that isn't right either.  I fixed both on master:
`error?: Error, block?: Block|null`
`error?: Error, logs?: Log[]`

A `null` `Block` is a valid non-error result, but if there are no logs then the result should be an empty `Log[]`.  In both cases either Error _or_ block/log must be defined and the other must be undefined (hence why they are both marked as optional).

**Pull request merged**